pkg/client/e2e.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/heptio/sonobuoy/pkg/results"
+	"github.com/pkg/errors"
+)
+
+// GetTests extracts the junit results from a sonobuoy archive and returns the requested tests.
+func (c *SonobuoyClient) GetTests(reader io.Reader, show string) (TestCases, error) {
+	read := results.NewReaderWithVersion(reader, "irrelevant")
+	junitResults := jUnitResults{}
+	err := read.WalkFiles(func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		return results.ExtractFileIntoStruct("plugins/e2e/results/junit_01.xml", path, info, &junitResults)
+	})
+	if err != nil {
+		return TestCases{}, errors.Wrap(err, "failed to walk results archive")
+	}
+	out := make(TestCases, 0)
+	if show == "passed" || show == "all" {
+		out = append(out, junitResults.filter(passed)...)
+	}
+	if show == "failed" || show == "all" {
+		out = append(out, junitResults.filter(failed)...)
+	}
+	if show == "skipped" || show == "all" {
+		out = append(out, junitResults.filter(skipped)...)
+	}
+	sort.Sort(out)
+	return out, nil
+}
+
+type TestCases []TestCase
```

Comment:
This almost seems like this belongs with the tarball utility library, and not really part of the external client.